### DISCUSSION TITLE
Firefox "Authorization Denied" after page refresh

### DISCUSSION THREAD

**Discussion by TByte007** [Keyfactor/ejbca-ce#509] [Q&A]
I have a very strange problem with Firefox. At first it asks for a client certificate and it (kind of) works. "ejbca/adminweb" opens without "CA Status" getting filled but it opens. Then if I refresh the page I get:

> Authorization Denied
> No client certificate was presented
> If you did not get prompted to select a client certificate, please check that you have the correct certificate.

And if I refresh the page again it opens (more or less) fine again! And it goes on in a cycle.
The same thing works fine in Chrome. Does anybody have an idea what is going on?
(Version : EJBCA 8.2.0.1 Community)
Regards.

**Comment by primetomas** [comment]
I've never seen it. I use Firefox exclusively.

**Reply by TByte007** [reply]
I use it excursively too on FreeBSD, Windows and Linux  and at least on FreeBSD and Windows it does the same thing (not tested on Linux yet). I'm dumbfounded how it's even possible but I tested it on 2 different Windows machines and a FreeBSD machine. One thing that could be a problem is that the server cert is expired but it is not a problem on the older EJBCA (Version : EJBCA 6.2.0 (r19221)) which I'm upgrading. The certificates are the same literally. Of course the other thing that could be a problem is my Wildfly config.

**Reply by primetomas** [reply]
Yeah, the TLS handshake is done by WildFly.

**Reply by primetomas** [reply]
Only times I've seen issues are when the browser has "cached" to not use client cert, so a cache clean/browser restart is needed.

**Reply by primetomas** [reply]
But of course, an expired cert...renew that and see if it goes away.

**Reply by TByte007** [reply]
The same cert works with the older version but on JBoss. And it has no problem with Chrome for some reason. Firefox reports no problem too. It's just that on every refresh it switches from working to not working and vice versa. Never seen anything like that before. 
PS: On FreeBSD and the second Windows there were no cached version. On the first opening of the page and it happens.

**Comment by FaysalMossi** [comment]
Hello everyone
I too am currently experiencing the same problem on both Firefox and Chrome browsers.
i use  EJBCA 7.11.0
